docs/source/federation-guide/graph-administration.mdx

```diff
@@ -0,0 +1,201 @@
+
+## Tooling for Data Graph Consumers
+
+GraphQL offers a client-centric approach to developing APIs, and this promise extends beyond designing queries that are purpose-built to meet client developer needs and drive product experiences. Both Apollo and members of the GraphQL community have created extensive client-side tooling for web and mobile to expedite development and help teams ship features faster. In turn, client developers can maximize the utility of the schema registry by adopting best practices when sending their requests to the data graph.
+
+### Consuming the Data Graph
+
+When a company shifts toward a consolidated data graph, client developers who may have previously juggled multiple GraphQL endpoints or other point-to-point APIs no longer need to jump through client-side hoops to query all of the data needed to render a view in an application. That said, with a centralized schema registry in place, a common set of standards should be established to structure how clients make requests to the data graph.
+
+At a minimum, there should be basic controls on what operations are executed against the data graph, and how those operations are structured. Tracing data becomes far more useful and actionable when clients include their names and versions as metadata. When using Apollo Client for web or mobile, you can specify `name` and `version` options that will automatically be translated into header values sent with every request  (specifically, the headers are called `apollographql-client-name` and `apollographql-client-version`, and they may be set manually for other GraphQL clients). Tools such as Apollo Studio can then use this operation trace metadata to help service developers conduct more effective schema validation.
+
+In addition to identifying themselves by name and version, clients should use [named operations](https://spec.graphql.org/June2018/#sec-Named-Operation-Definitions) for each request to the data graph. For example, at Apollo, we prepend `UI_` to the names of all operations sent from the Apollo Studio web application. Other teams go so far as to add linting rules to check the structure of client operations as a part of that client's deployment pipeline too.
+
+While there isn't a single "best" way to structure operation names, the important takeaway here is to establish some operation-related standards, communicate those standards to teams, and then enforce them. Enforcement of these rules can take place within the Apollo Gateway, but adding runtime logic on a per-request basis should be approached with caution when rules may be checked statically in client code as part of the CI/CD process instead. However enforced, both client awareness and operation names are essential in providing the necessary visibility in observability tools to support a field deprecation and rollover strategy that prevents breaking changes for client developers as the data graph evolves.
+
+### Code Generation
```

Comment:
@hwillson is there anything we want to tweak here about codegen? In our current state we don't recommend the `apollo client:codegen` and just wanted to double check with you

Reply:
We're all good here @michael-watson - @mandiwise reached out to me about this, and we landed on recommending `graphql-code-generator` (for now anyways). If we ever decide to double down on TS codegen in the future, we can change our recommendation, but for now our solution is way behind gcg (and we're planning on deprecating it). Thanks for pinging me on this!